src/nbd.c

```diff
@@ -556,12 +556,12 @@ static void start_configure(struct RaucNBDContext *ctx, struct RaucNBDTransfer *
 	gboolean res = FALSE;
 	CURLcode code = 0;
 	CURLMcode mcode = 0;
-	g_autofree guint8 *data = g_malloc(xfer->request.len);
-	g_autoptr(GVariant) v = NULL;
-	GVariantDict dict;
 
 	/* only read from the client on the first try */
 	if (!ctx->url) {
+		g_autofree guint8 *data = g_malloc(xfer->request.len);
+		g_autoptr(GVariant) v = NULL;
+		g_auto(GVariantDict) dict = G_VARIANT_DICT_INIT(NULL);
```

Comment:
Maybe note that this changes from stack-allocated to heap-allocated and that the former variant lacked the `g_variant_dict_clear()` call?

Also, the commit contains an unrelated newline change and unrelated code move. But since the result is still clear enough to read I'd be fine with that.

Reply:
This is still stack-allocated, which is what `g_auto` is used for.

Reply:
Ouh, right. My fault.